### DISCUSSION TITLE
有关ios最新版的一些问题

### DISCUSSION THREAD

**Discussion by unauth401** [Lost-MSth/Arcaea-server#157] [Q&A]
目前涉及到的设备有：
iPad iOS v16.7.1 Arcaea v5.5.6 已添加并信任链路上的根证书和自签名证书
笔记本 mitmproxy 转发api流量到服务器 开启--ssl-insecure
服务器 linux debian12 Arcaea-server 用nginx反代了 openssl自签名证书

网络/设备拓扑目前是这样：
iPad和笔记本在同一WiFi下（同一LAN内）
所有三个设备在一个tailscale局域网下
iPad通过笔记本上的mitmproxy转发流量到服务器，目前iPad上的Safari可以不弹`不安全的证书`相关提示请求到Arcaea-server。

但目前客户端内只要涉及到api的内容，请求一发就从客户端断开连接，查了一圈发现可能是ssl-pinning的问题。
但是直接用压缩软件解包没有看到非常”明显“的证书文件以及解密密码（好久之前整过一个安卓的ssl-pinning，那个有证书文件以及硬编码在代码中的明文私钥）
也找到一篇说你自己解包然后patch再自签名的安装包，但这东西好想要氪一发Apple Developer，688有点伤。而且重新打包的话似乎跟原版就是两个存档了吧（哥们目前有一个给自己发9999碎片的需求所以...）

想问问各位我下一步应该怎么做，是接着尝试提取证书，还是尝试重新打包。谢谢。

**Comment by Lost-MSth** [comment]
可以ā，都弄捯这了不僦差ー丶吗。提取证書侑甚么用，沒囿服雾端证书也沒法伪造ā（

正黍和幂碼全石更編碼加米在那箇二進制中了，你应该关註些别的，仳茹怎么坡鋽 SSL pinning。你砢能砢以找捯ー两箇阈名，攺鋽它说不定囿用。其它地方好像还要做些甚么来着，攺法不唯ー。

攺二進制后肯定要重新打包，iOS 砢以侧載来着，或者还囿箇 TrollStore，那僦不需要开發者賬号了。

艹我自已看都要2333④了（）

**Reply by unauth401** [reply]
操笑死了哈哈哈哈哈哈哈哈哈
多谢了 明天试试
